token/program-2022/src/instruction.rs

```diff
@@ -409,6 +418,102 @@ pub enum TokenInstruction {
+    ///   1. `[]` The token mint. Must be a MintWithTransferFee.
+    ///   2. `[writable]` The destination account. Must be an AccountWithTransferFee.
+    ///   3. `[writable]` The fee receiver account. Must be owned by the mint's
+    ///      `fee_account_owner`.
+    ///   4. `[signer]` The source account's owner/delegate.
+    ///
+    ///   * Multisignature owner/delegate
+    ///   0. `[writable]` The source account.
+    ///   1. `[]` The token mint.
+    ///   2. `[writable]` The destination account.
+    ///   3. `[writable]` The fee receiver account. Must be owned by the mint's
+    ///      `fee_account_owner`.
+    ///   4. `[]` The source account's multisignature owner/delegate.
+    ///   5. ..5+M `[signer]` M signer accounts.
+    TransferWithFee {
+        /// The amount of tokens to transfer.
+        amount: u64,
+        /// Expected number of base 10 digits to the right of the decimal place.
+        decimals: u8,
+    },
+    /// Transfer withheld tokens to a fee account. Signed by either the source
+    /// account owner or the fee account owner.
+    ///
+    /// Succeeds for frozen accounts. <-- Does this make sense?
+    ///
+    /// Note: this could be made permissionless, but that may open up attack
+    /// opportunities.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single owner/delegate
+    ///   0. `[writable]` The source account. Must be an AccountWithTransferFee.
+    ///   1. `[]` The token mint. Must be a MintWithTransferFee.
+    ///   2. `[writable]` The fee receiver account. Must be owned by the mint's
+    ///      `fee_account_owner`.
+    ///   3. `[signer]` The source account's owner or the mint's `fee_account_owner`
+    ///
+    ///   * Multisignature owner/delegate
+    ///   0. `[writable]` The source account.
+    ///   1. `[]` The token mint.
+    ///   2. `[writable]` The destination account.
+    ///   3. `[]` The source or destination account's multisignature owner/delegate.
+    ///   4. ..4+M `[signer]` M signer accounts.
+    TransferWithheld {
+        /// The amount of tokens to transfer.
+        amount: u64,
+        /// Expected number of base 10 digits to the right of the decimal place.
+        decimals: u8,
+    },
+    /// Set transfer fee. Only supported for `MintWithTransferFee`s.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single authority
+    ///   0. `[writable]` The mint.
+    ///   1. `[signer]` The mint's fee account owner.
+    ///
+    ///   * Multisignature authority
+    ///   0. `[writable]` The mint.
+    ///   1. `[]` The mint's multisignature fee account owner.
+    ///   2. ..2+M `[signer]` M signer accounts.
+    SetTransferFee {
+        /// Amount of transfer collected as fees, expressed as basis points of the
+        /// transfer amount
+        transfer_fee_basis_points: u16,
+        /// Maximum fee assessed on transfers
+        maximum_fee: u64,
```

Comment:
```suggestion
        maximum_fee: Option<u64>,  /// `None` means there's no maximum fee
```

Reply:
I was thinking it could be set to u64::MAX for no max fee, but Option is probably clearer